bootstrap/kubeadm/internal/locking/control_plane_init_mutex.go

```diff
@@ -62,16 +62,25 @@ func (c *ControlPlaneInitMutex) Lock(ctx context.Context, cluster *clusterv1.Clu
 	case err != nil:
 		log.Error(err, "Failed to acquire lock")
 		return false
-	default: // successfully found an existing config map
+	default: // Successfully found an existing config map.
 		info, err := sema.information()
 		if err != nil {
 			log.Error(err, "Failed to get information about the existing lock")
 			return false
 		}
-		// the machine requesting the lock is the machine that created the lock, therefore the lock is acquired
+		// The machine requesting the lock is the machine that created the lock, therefore the lock is acquired.
 		if info.MachineName == machine.Name {
 			return true
 		}
+
+		// If the machine that created the lock can not be found unlock the mutex.
+		err = c.client.Get(ctx, client.ObjectKey{
+			Namespace: cluster.Namespace,
+			Name:      info.MachineName,
+		}, &clusterv1.Machine{})
+		if apierrors.IsNotFound(err) {
```

Comment:
```suggestion
		if err := c.client.Get(ctx, client.ObjectKey{
			Namespace: cluster.Namespace,
			Name:      info.MachineName,
		}, &clusterv1.Machine{}); apierrors.IsNotFound(err) {
```

Let's also handle the else if err != nil case?

Reply:
I don't think we want to block on any other type of error - but we could log if there's some issue.

Reply:
Yeah, that's what I was thinking earlier, a different branch that returns false and logs

Reply:
The new version always logs the error and unlocks only if the err is apierrors not found.